tests/kafkatest/services/security/minikdc.py

```diff
@@ -65,7 +65,7 @@ def start_node(self, node):
         self.logger.info(props_file)
 
         kafka_principals = ' '.join(['kafka/' + kafka_node.account.hostname for kafka_node in self.kafka_nodes])
-        principals = 'client ' + kafka_principals + self.extra_principals
+        principals = 'client ' + kafka_principals + ' ' + self.extra_principals
```

Comment:
`extra_principals` is currently never used. I found it confusing that I was expected to set `extra_principals` with a leading `' '`. This isn't needed, but IMO it cleans up the API.